File: frontend/lib/features/staff_leave/data/models/leave_type.dart
enum LeaveType {
  casual('casual', 'Casual Leave'),
  medical('medical', 'Medical Leave'),
  earned('earned', 'Earned Leave'),
  halfDay('half_day', 'Half Day');

  const LeaveType(this.apiValue, this.label);

  final String apiValue;
  final String label;

  static LeaveType fromApiValue(String value) => LeaveType.values.firstWhere((t) => t.apiValue == value);
}

File: backend/app/Enums/LeaveType.php
<?php

namespace App\Enums;

enum LeaveType: string
{
    case Casual = 'casual';
    case Medical = 'medical';
    case Earned = 'earned';
    case HalfDay = 'half_day';

    public function label(): string
    {
        return match ($this) {
            self::Casual => 'casual',
            self::Medical => 'medical',
            self::Earned => 'earned',
            self::HalfDay => 'half-day',
        };
    }
}
